classes/LunchBot.js

```diff
@@ -129,23 +128,42 @@ class LunchBot {
 
 
 	addReactions() {
-		const self = this;
-
-		return new Promise(resolve => {
-			/**
-			 * Posts a reaction beneath the options
-			 * @param {string} reaction
-			 * @param {callback} done
-			 */
-			function postReaction(reaction, done) {
-				self.bot.api.callAPI("reactions.add", {
-					channel: self.pollMessage.channel,
-					timestamp: self.pollMessage.message.ts,
-					user: self.bot.identity.id,
-					name: reaction
-				}, done);
+		const api = this.bot.api;
+		const controller = this.botkitController;
+		const params = {
+			channel: this.pollMessage.channel,
+			timestamp: this.pollMessage.ts,
+			user: this.bot.identity.id
+		};
+		const iterator = REACTIONS[Symbol.iterator]();
+		let position = iterator.next();
+
+		const onnext = (resolve, reject) => {
+			if (position.done) {
+				return resolve();
 			}
-			async.eachSeries(REACTIONS, postReaction, resolve);
+
+			params.name = position.value;
+			api.callAPI("reactions.add", params, err => {
+				position = iterator.next();
+
+				if (err) {
+					reject(err);
+					position.done = true;
+				}
+			});
+		};
```

Comment:
I can make this function slightly more readable at the cost of needing to create a callback that delegates to resolve/reject in the promise like the following.

```javascript
new Promise((resolve, reject) => {
   const callback = (err, done) => {
      if (err) return reject(err);
      if (done) return resolve();
   };

   // ...
});
```

Do you think that is preferable?